public/static/js/distros.js

```diff
@@ -602,10 +666,13 @@ mciModule.controller('DistrosCtrl', function ($scope, $window, $http, $location,
   // if a security group is in a vpc it needs to be the id which starts with 'sg-'
   $scope.validSecurityGroup = function () {
     if ($scope.activeDistro) {
-      if ($scope.activeDistro.settings.is_vpc && $scope.activeDistro.settings.security_group_ids) {
-        for (var i = 0; i < $scope.activeDistro.settings.security_group_ids.length; i++) {
-          if ($scope.activeDistro.settings.security_group_ids[i].substring(0, 3) !== "sg-") {
-            return false
+      $scope.updateSettingsList(); // to validate list
```

Comment:
Both these functions are called by the same function, so you end up updating the list twice. I'd move this into validForm

Reply:
Done